docs/contributing/git_cheatsheet.md

```diff
@@ -0,0 +1,308 @@
+
+> Git Cheatsheet for stdlib
+
+## Introduction
+
+Welcome to `stdlib`! [Git][git] can feel overwhelming at times, but don't worry! This cheatsheet will walk you through the essential Git commands you'll need in a structured and easy-to-follow manner. Think of it as your go-to guide for a smooth and hassle-free workflow.
+
+## Configuration
+
+Before you start using [Git][git], it's important to introduce yourself. This step ensures that your commits are linked to your identity, making it easier to track contributions.
+
+<!-- run-disable -->
+
+```bash
+git config --global user.name "Your Name"
+git config --global user.email "your.email@example.com"
+```
+
+> You only need to do this once.
+
+To confirm that your identity is set correctly, run:
+
+<!-- run-disable -->
+
+```bash
+git config --global user.name
+git config --global user.email
+```
+
+This should display the name and email you configured. If there's a mistake, simply re-run the configuration commands with the correct details.
+
+## Cloning
+
+To contribute to `stdlib`, you first need to [fork][github-fork] the repository on GitHub. This creates a copy of the repository under your GitHub account where you have write access.
+
+Once you've forked the repository, clone your fork onto your local machine:
+
+<!-- run-disable -->
+
+```bash
+git clone https://github.com/YOUR_GITHUB_USERNAME/stdlib.git
+cd stdlib
+```
+
+Since the official `stdlib` repository keeps updating, link it as the [upstream][git-remotes] remote to fetch the latest changes:
+
+<!-- run-disable -->
+
+```bash
+git remote add upstream https://github.com/stdlib-js/stdlib.git
+git fetch upstream
+```
+
+> In [Git][git], a [remote][github-remote] is a reference to a repository. Your fork is called `origin`, while the official `stdlib` repository is referred to as `upstream`. Adding an upstream remote allows you to consistently fetch the latest updates from the original repository and incorporate them into your work.
+
+## Branching
+
+In `stdlib`, the `develop` branch is the primary branch for development. Instead of making changes directly in the `develop` branch of your forked repository, it's best to create a separate branch for each feature or fix. This keeps your `develop` branch clean and makes it easier to pull updates from the official repository. To create a new branch for your work, use:
+
+<!-- run-disable -->
+
+```bash
+git checkout -b feature/my-new-feature
+```
+
+## Changes
+
+Now comes the fun part of actually writing code! After making your changes, it's always good to check what’s modified:
+
+<!-- run-disable -->
+
+```bash
+git status
+```
+
+Once you're happy with your changes, add them to the staging area for a final confirmation:
+
+<!-- run-disable -->
+
+```bash
+git add <file1> <file2>  # Add specific files
+git add .                # Add all changes
+```
+
+Then, commit with a meaningful message:
+
+<!-- run-disable -->
+
+```bash
+git commit -m "feat: add support for new function"
```

Comment:
@anandkaranubc I think it worthwhile to also include a commit example using `git commit`. Multi-line commits have tripped people up in the past.